dockers/docker-fpm-frr/docker_init.sh

```diff
@@ -6,6 +6,23 @@ mkdir -p /etc/supervisor/conf.d
 sonic-cfggen -d -t /usr/share/sonic/templates/supervisord/supervisord.conf.j2 > /etc/supervisor/conf.d/supervisord.conf
 
 CONFIG_TYPE=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["docker_routing_config_mode"]'`
+
+if [[ ! -z "$NAMESPACE_ID" ]]; then
+   # FRR is not running in host namespace so we need to delete
+   # default gw kernel route added by docker network via eth0 and add it back 
+   # with higer administrative distance so that default route learn
+   # by FRR becomes best route if/when available
+   GATEWAY_IP=$(ip route show 0.0.0.0/0 dev eth0 | awk '{print $3}')
+   ip route del 0.0.0.0/0 dev eth0
+   # Ref: http://docs.frrouting.org/en/latest/zebra.html#zebra-vrf
+   # Zebra does treat Kernel routes as special case for the purposes of Admin Distance. \
+   # Upon learning about a route that is not originated by FRR we read the metric value as a uint32_t. 
+   # The top byte of the value is interpreted as the Administrative Distance and 
+   # the low three bytes are read in as the metric.
+   # so here we are programming administrative distace of 210 (210 << 24) > 200 (for routes learn via IBGP)
+   ip route add 0.0.0.0/0 via $GATEWAY_IP dev eth0 metric 3523215360 
```

Comment:
is it possible to modify the route, instead of removing/adding it?

Reply:
@pavel-shirshov i tried two option:

- ip route change (This command fails)

- ip route replace (It creates one more new default route with metric)

so i feel cleaner is just to delete and add it back.

Reply:
I think you need ip route change command. What error message do you see when you apply this command?
Why do you think that remove/add is cleaner?

Reply:
@pavel-shirshov :  
On using route change getting error "RTNETLINK answers: No such file or directory"
ip route change 0.0.0.0/0 via 240.127.1.1  dev eth0 metric 200
RTNETLINK answers: No such file or directory

My point for cleaner was w.r.t to "ip route replace" since it creates one more new default route with metric so we will have two routes. 
If we del/add then there is only one and it's cleaner in my opinion.

Reply:
@pavel-shirshov Can you please review this. I have address your comments.

Reply:
@abdosi I think it is better to "change" or check that the route was removed. Otherwise we have a chance to introduce another route without removing the previous one.

Reply:
@pavel-shirshov added check to add only if docker default route exist and delete is successful